server/lib/update-linked-fields.js

```diff
@@ -1,54 +1,62 @@
 import Customer from '../models/customer'
 import Volunteer from '../models/volunteer'
 import Donor from '../models/donor'
+import {clientRoles} from '../../common/constants'
 
 /**
-* Update Volunteer fields of address
+* Update fields of address
 */
-export async function updateVolunteer(reqbodyfields, user_id) {
+ async function update(Model, reqbodyfields, user_id) {
   //Updating address of the volunteer role if exists
-  const volunteer = await Volunteer.findById(user_id)
-  if(volunteer != undefined && volunteer != null && volunteer != {}){
-    let oldAdrFields = volunteer.fields
-    for(let j = 0; j < oldAdrFields.length; j++){
-      for(let k = 0; k < reqbodyfields.length; k++){
-        if(oldAdrFields[j].meta == reqbodyfields[k].meta) oldAdrFields[j].value = reqbodyfields[k].value
+  const model = await Model.findById(user_id)
+  if (model != undefined && model != null && model != {}){
+    let oldAdrFields = model.fields
+    for (let j = 0; j < oldAdrFields.length; j++){
+      for (let k = 0; k < reqbodyfields.length; k++){
+        if (oldAdrFields[j].meta == reqbodyfields[k].meta) oldAdrFields[j].value = reqbodyfields[k].value
       }
     }
-    await Volunteer.findByIdAndUpdate(user_id, {fields: oldAdrFields})
+    await Model.findByIdAndUpdate(user_id, {fields: oldAdrFields})
   }
 }
 
 /**
+* Update Volunteer fields of address
+*/
+ async function updateVolunteer(reqbodyfields, user_id) {
```

Comment:
these functions don't have much value any more, you can just call `update` directly in the `updateFields` function